### PR TITLE
Refactor: Remove docs which overlap with Flux website

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- Explicitly call kustomize-controller a Flux component
- Remove guide form README.md which overlaps with Flux website
- Add links to Flux website
- Make small corrections to the API docs

Ref: #663 #664

### CODE REVIEW

File: docs/spec/v1beta2/kustomization.md [hunk 1]
```diff
@@ -201,23 +204,22 @@ spec:
...
 ```sh
-cd clusters/my-cluster
+cd apps/my-app
 
-# create kustomization
+# create kustomization.yaml
 kustomize create --autodetect --recursive
 
-# validate kustomization
+# validate all manifests
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply docs/spec/v1beta2/kustomization.md]: I have removed the validation step, we have this in FAQ where we explain our usage of kubeconform.

**Review by hiddeco — APPROVED** [review]
I do not think that this closes #664, as it is still structured in a different way than the new spec documentation around sources (e.g. "Writing a Kustomization spec" like https://github.com/fluxcd/source-controller/blob/main/docs/spec/v1beta2/buckets.md#writing-a-bucket-spec, and explicit mention of fields and their behavior).

**Comment by stefanprodan** [comment]
> I do not think that this closes #664

It doesn't, I just added a ref to it.

**Comment by hiddeco** [comment]
Ah, absolutely my bad for misreading.

**Pull request merged**